### ISSUE TITLE
Problems encountered during evaluation of pretrained models and training

### ISSUE THREAD

**Issue by tianjian00** [Wireless3C-SJTU/MambaJSCC#8]
Dear Author,

Thank you very much for open-sourcing this excellent work.
I am trying to reproduce your results as a baseline for comparison. However, during reproduction I was only able to obtain part of the expected results, while some results remain inconsistent with those reported.
Below, I describe my procedures and encountered issues in detail, and I would greatly appreciate your help and advice.

1. Problems encountered during evaluation of pretrained models

During testing with pretrained models, all fixed-SNR models where CHANNEL.ADAPTIVE = no produced the expected results.
However, for pretrained models with CHANNEL.ADAPTIVE = ssm, the evaluation results appeared abnormal.

For example:
Using your pretrained models
encoderOUTCHANS32_extentMLP_lossPSNR_SCANnum2_SNR[1, 3, 5, 7, 9, 11, 13, 15, 17, 20]_adpssm_typerayleigh_depth4_embed[128, 192, 256, 320]_nums[2, 2, 6, 2]_rsl256.pt
decoderOUTCHANS32_extentMLP_lossPSNR_SCANnum2_SNR[1, 3, 5, 7, 9, 11, 13, 15, 17, 20]_adpssm_typerayleigh_depth4_embed[128, 192, 256, 320]_nums[2, 2, 6, 2]_rsl256.pt

When testing on the Kodak dataset under a Rayleigh channel with SNR = [1, 10, 20] dB, the obtained PSNR values were:
[24.868, 23.524, 22.772] dB, which seem significantly lower than expected.

As comparison：

① pretrained models
encoderOUTCHANS32_extentMLP_lossPSNR_SCANnum2_SNR[1, 4, 7, 10, 13]_adpssm_typerayleigh_depth4_embed[128, 192, 256, 320]_nums[2, 2, 6, 2]_rsl256.pt
decoderOUTCHANS32_extentMLP_lossPSNR_SCANnum2_SNR[1, 4, 7, 10, 13]_adpssm_typerayleigh_depth4_embed[128, 192, 256, 320]_nums[2, 2, 6, 2]_rsl256.pt

When testing on the Kodak dataset under a Rayleigh channel with SNR = [1, 10] dB, the obtained PSNR values were:
[7.9093, 7.9034] dB, which is actually no image reconstructed

② pretrained models
encoderOUTCHANS115_extentMLP_lossPSNR_SCANnum2_SNR[10]_adpno_typeawgn_depth4_embed[128, 192, 256, 320]_nums[2, 2, 6, 2]_rsl256_CLIC2021.pt
decoderOUTCHANS115_extentMLP_lossPSNR_SCANnum2_SNR[10]_adpno_typeawgn_depth4_embed[128, 192, 256, 320]_nums[2, 2, 6, 2]_rsl256_CLIC2021.pt

When testing on the Kodak dataset under an AWGN channel with SNR = [10] dB, the obtained PSNR values were:
[34.7064] dB,

The model loading code was as follows:
encoder_path = 'path_to_encoder_ckpt'
decoder_path = 'path_to_decoder_ckpt'
encoder = torch.load(encoder_path, weights_only=False)
decoder = torch.load(decoder_path, weights_only=False)

I am using an NVIDIA RTX 3090 GPU, and the following are my installed libraries and their versions:
pip list
Package                 Version
aiohappyeyeballs        2.6.1
aiohttp                 3.12.13
aiosignal               1.3.2
annotated-types         0.7.0
antlr4-python3-runtime  4.9.3
async-timeout           5.0.1
attrs                   25.3.0
Automat                 25.4.16
buildtools              1.0.6
causal-conv1d           1.1.1
certifi                 2022.12.7
charset-normalizer      2.1.1
click                   8.2.1
constantly              23.10.4
docopt                  0.6.2
einops                  0.8.1
filelock                3.13.1
frozenlist              1.7.0
fsspec                  2024.6.1
furl                    2.1.4
fvcore                  0.1.5.post20221221
gitdb                   4.0.12
GitPython               3.1.44
greenlet                3.2.3
hf-xet                  1.1.5
huggingface-hub         0.33.1
hyperlink               21.0.0
idna                    3.4
incremental             24.7.2
iopath                  0.1.10
Jinja2                  3.1.4
lightning-utilities     0.14.3
mamba-ssm               1.1.3.post1
MarkupSafe              2.1.5
mpmath                  1.3.0
multidict               6.6.2
networkx                3.3
ninja                   1.11.1.4
numpy                   1.26.4
omegaconf               2.3.0
opencv-python           4.12.0.88
orderedmultidict        1.0.1
packaging               24.2
pillow                  11.0.0
pip                     25.1
platformdirs            4.3.8
portalocker             3.2.0
propcache               0.3.2
protobuf                6.31.1
psutil                  7.0.0
pydantic                2.11.7
pydantic_core           2.33.2
python-dateutil         2.9.0.post0
pytorch-lightning       2.5.2
PyYAML                  6.0.2
redo                    3.0.0
regex                   2024.11.6
requests                2.28.1
safetensors             0.5.3
sentry-sdk              2.32.0
setproctitle            1.3.6
setuptools              78.1.1
simplejson              3.20.1
six                     1.17.0
smmap                   5.0.2
SQLAlchemy              2.0.41
sympy                   1.13.3
tabulate                0.9.0
taming-transformers     0.0.1
termcolor               3.1.0
timm                    0.4.12
tokenizers              0.21.2
tomli                   2.2.1
torch                   2.1.1+cu118
torch-operation-counter 0.3.4
torchaudio              2.1.1+cu118
torchmetrics            1.7.3
torchvision             0.16.1+cu118
tqdm                    4.67.1
transformers            4.53.0
triton                  2.1.0
Twisted                 25.5.0
typing_extensions       4.12.2
typing-inspection       0.4.1
urllib3                 1.26.13
wandb                   0.20.1
wheel                   0.45.1
yacs                    0.1.8
yarl                    1.20.1
zope.interface          7.2

2. Problems encountered during training

I also encountered some issues during training.
The hardware and environment configurations were the same as above.
Specifically, the training loss decreased for a while and then plateaued, while the trained model still performed noticeably worse than your pretrained model.
Even after increasing the number of epochs, there was no significant improvement (the loss function used was MSELoss).
<img width="640" height="480" alt="Image" src="https://github.com/user-attachments/assets/1feb7955-0da6-4806-8731-a8e6ca28a696" />

My training configuration is as follows:
train:
TRAIN:
  EPOCHS: 2000
  SAVE_FRE: 100
  LOSS: 'PSNR'        ## PSNR, MS-SSIM, LPIPS
  EVAL_MATRIX: 'PSNR' ## PSNR, MS-SSIM, LPIPS
  GAN_LOSS: False
DATA:
  DATASET: "DIV2K"
  TRAIN_BATCH: 4
  train_data_dir: "path_to_DIV2K_train_HR"
CHANNEL:
  TYPE: 'rayleigh'
  SNR : [10]
  ADAPTIVE: 'ssm'

vssm:
MODEL:
  TYPE: vssm
  NAME: vssm_tiny
  DROP_PATH_RATE: 0.2
  VSSM:
    EMBED_DIM: [128,192,256,320]
    DEPTHS: [2, 2, 6, 2]
    OUT_CHANS: 96
    SCAN: 'cross'
    PE: 'no'
    SCAN_NUMBER: 2
    Extent: 'MLP'
    channel_input: 'conv'

If you have any suggestions on possible causes for these issues or recommendations for resolving them, I would be sincerely grateful for your guidance.

Thank you again for your excellent contribution and for making your work publicly available.

Best regards

**Comment by tianjian00** [comment]
I’m not sure what went wrong. Thanks a lot for your help! Please let me know if you need any further details.
